app/models/miq_region.rb

```diff
@@ -289,6 +293,15 @@ def self.display_name(number = 1)
     n_('Region', 'Regions', number)
   end
 
+  def reload(*_, &block)
+    clear_my_relation_cache
+    super
```

Comment:
Also seen this written elsewhere as 

```suggestion
    super.tap { clear_my_relation_cache }
```